### PR TITLE
Add mountpoint functionality

### PR DESCRIPTION
**Pull request by solidgoldpig**
If a pageInstance has a `mountPoint` property, it should:

- use the mountPoint instance's URL as its potential base URL
- inherit any show conditions from the mountPoint instance
- inherit any namespace from the mountPoint instance

### CODE REVIEW

File: lib/propagate-categories.js [hunk 1]
```diff
@@ -15,7 +18,7 @@ const {deepClone} = require('@ministryofjustice/fb-utils-node')
  *  Object of service schemas keyed by _name
  *
  * @return {object}
- *  Cloned object containing instances with propagated namespaces
+ *  Cloned object containing instances with propagated categorys
```
Comment (umaar) [inline lib/propagate-categories.js]: Typo

File: lib/propagate-show.js [hunk 2]
```diff
@@ -93,6 +95,20 @@ const propagate = instances => {
...
      **/
   const propagateStepConditions = (instance) => {
+    if (seen[instance._id]) {
+      return
+    }
+    seen[instance._id] = true
+    if (instance.mountPoint) {
+      const mountPointInstance = instances[instance.mountPoint]
+      propagateStepConditions(mountPointInstance)
+      if (mountPointInstance.show !== undefined) {
```
Comment (umaar) [inline lib/propagate-show.js:105]: Just a minor thought, the property value should never be undefined right? But rather, the property may not be present? If so, `Object.prototype.hasOwnProperty` could be nice in terms of semantics

...a very minor point however

File: lib/propagate-steps.js [hunk 3]
```diff
@@ -150,7 +162,15 @@ const propagateNamespaces = (instances, instance) => {
...
  **/
 const propagateUrls = (instances, instance) => {
-  const serviceRoot = '/'
+  let serviceRoot = ''
+  if (instance.mountPoint) {
+    const mountPointInstance = instances[instance.mountPoint]
+    propagateUrls(instances, mountPointInstance)
+    serviceRoot = instances[instance.mountPoint].url
+  }
+  if (!serviceRoot.endsWith('/')) {
```
Comment (umaar) [inline lib/propagate-steps.js:171]: How come?

**Review by umaar — APPROVED** [review]
Nice

File: lib/propagate-categories.js (hunk 1 above)
Comment (solidgoldpig) [inline lib/propagate-categories.js]:
```suggestion
Cloned object containing instances with propagated categories
```

Comment (solidgoldpig) [reply lib/propagate-categories.js]: Fixed

Comment (solidgoldpig) [reply lib/propagate-categories.js]: Fixed

File: lib/propagate-show.js (hunk 2 above)
Comment (solidgoldpig) [reply lib/propagate-show.js:105]: Go for it

File: lib/propagate-steps.js (hunk 3 above)
Comment (solidgoldpig) [reply lib/propagate-steps.js:171]: If a page has no parent and its url isn't absolute, we need to add its url to either the root url or the mountPoint instance's url.

The root url is obviously '/', but the mountInstance will be something like '/mountpoint'

This just ensures there's a slash between the 2 parts

**Pull request merged**